api/clients/v2/payloaddispersal/payload_disperser.go

```diff
@@ -20,21 +22,24 @@ type PayloadDisperser struct {
 	config          PayloadDisperserConfig
 	disperserClient clients.DisperserClient
 	certVerifier    clients.ICertVerifier
+	stageTimer      *common.StageTimer
 }
 
 // NewPayloadDisperser creates a PayloadDisperser from subcomponents that have already been constructed and initialized.
+// If the registry is nil then no metrics will be collected.
```

Comment:
this doc would seem better placed with the `registry` param

Reply:
```
	// if nil, then no metrics will be collected
	registry *prometheus.Registry,
```